code/modules/antagonists/clockcult/clock_structures/ratvar_the_clockwork_justicar.dm

```diff
@@ -151,3 +152,35 @@
 			sound_to_playing_players('sound/machines/clockcult/ratvar_scream.ogg', 80)
 			narsie.clashing = FALSE
 			qdel(src)
+
+
+/obj/structure/destructible/clockwork/massive/ratvar/proc/purge_the_heresy()
+	sleep(50)
+	priority_announce("Massive energy surge detected. Closest matching threat: Incoming supernova. All crew are advised to evacuate NAN lightyears away from blast zone","Central Command Higher Dimensional Affairs", 'sound/misc/airraid.ogg')
+	sleep(300)
+	priority_announce("Gravitational anomalies detected on the station. Th$%e %o a& ad$&%al d%&-BZZZZZT.","Central Command Higher Dimensional Affairs", 'sound/magic/clockwork/ratvar_announce1.ogg')
+	sleep(80)
+	sound_to_playing_players(sound('sound/magic/clockwork/ratvar_announce2.ogg'))
+	send_to_playing_players("<span class='heavy_brass'><font size=5>\"COME, ALL THOSE FAITHFUL! WITNESS THE RAYS OF JUSTICE CAST UPON THE HERETICS!\"</font></span>")
+	sleep(50)
+	set_security_level("delta")
+	SSshuttle.registerHostileEnvironment(src)
+	SSshuttle.lockdown = TRUE
+	sleep(250)
+	if(QDELETED(src))
+		priority_announce("Energy signal no longer detected.","Central Command Higher Dimensional Affairs")
+		return
+	sound_to_playing_players('sound/machines/clockcult/ratvar_scream.ogg', 80) //if this isn't lessened in volume it peaks for some reason
+	addtimer(CALLBACK(GLOBAL_PROC, /proc/clockcult_ending_helper), 300)
+
+/proc/clockcult_ending_helper()
+	for(var/mob/M in GLOB.mob_list)
+		if(M.client)
+			SEND_SOUND(M, sound('sound/magic/clockwork/ratvar_attack.ogg'))
+			SEND_SOUND(M, sound('sound/magic/clockwork/ratvarfire.ogg'))
+		if(!is_servant_of_ratvar(M) && isliving(M))
+			var/mob/living/L = M
+			L.fire_stacks = INFINITY
+			L.IgniteMob()
+		sleep(50)
```

Comment:
Is this intended to wait 5 seconds after each mob it handles? That could take quite alot of time when there's a bunch of players / simplemobs.

Reply:
It sets every living mob on fire, then ends the round after five seconds to give people at least _some_ time to go "oh god I'm on fire"

Reply:
Should be on the same level as the for then, instead of *in* the for loop